File: Algorithm_171_Excel Sheet Column Number.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;
int titleToNumber(string s)
{
	int len=s.length();
    int res=0;
    for (int i=0;i<len;i++) res+=(s[i]-'A'+1)*pow(26,len-i-1);
    return res;
}
int main()
{
	string s="AZ";
	cout<<titleToNumber(s);
	return 0;
}

File: Algorithm_171_Excel Sheet Column Number.java
// 思路：本质上是26进制

class Solution {
    public int titleToNumber(String s) {
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            res *= 26;
            res += s.charAt(i) - 'A' + 1;
        }
        return res;
    }
}